### PR TITLE
ddl: support read generated columns with copr for adding index

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #35983

Problem Summary:

In #39191, I left a [TODO](https://github.com/pingcap/tidb/pull/39191/files#diff-a806101b581c9a964a55717c28432c7d094b6b810e8fa7c4f64ab54d23a8a225R84) about the support of reading the generated column with coprocessor.

### What is changed and how it works?

This PR supports reading the generated columns with coprocessor during adding index.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: ddl/backfilling.go [hunk 1]
```diff
@@ -744,9 +744,9 @@ func (b *backfillScheduler) initCopReqSenderPool() {
 		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender", zap.Error(err))
 		return
 	}
-	copCtx := newCopContext(b.tbl.Meta(), indexInfo, sessCtx)
-	if copCtx == nil {
-		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender")
+	copCtx, err := newCopContext(b.tbl.Meta(), indexInfo, sessCtx)
+	if err != nil {
+		logutil.BgLogger().Warn("[ddl-ingest] cannot init cop request sender", zap.Error(err))
```
Comment (hawkingrei) [inline ddl/backfilling.go:749]: Should be Error level ?

Comment (tangenta) [reply ddl/backfilling.go:749]: If the initialization failed, it will fallback to read record with the txn method.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**